#!/usr/bin/env python
PACKAGE = "autopilot_monitor"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# autopilot_monitor
gen.add("enable_param_change",   bool_t,   0, "",  False)
gen.add("goal_x", double_t, 0, "", 0, -10.0, 10.0)
gen.add("goal_y", double_t, 0, "", 0, -10.0, 10.0)
gen.add("goal_z", double_t, 0, "", 3, 0.0, 10)
gen.add("goal_heading", double_t, 0, "", 0, -180.0, 180.0)
gen.add("max_altitude", double_t, 0, "", 5, 0.0, 10.0)
gen.add("max_normalized_thrust", double_t, 0, "", 1.0, 0.0, 1.0)
gen.add("thrust_scale", double_t, 0, "", 1.0, 0.0, 1.0)

gen.add("PX4_POSITION_CTL",   bool_t,   0, "",  False)
gen.add("MPC_POSITION_CTL",   bool_t,   0, "",  False)
gen.add("Q_pos_xy", int_t, 0, "", 20, 0, 500)
gen.add("Q_pos_z", int_t, 0, "", 20, 0, 500)
gen.add("R_thrust", int_t, 0, "", 1, 0, 100)
gen.add("R_pitchroll", int_t, 0, "", 100, 0, 200)
gen.add("R_yaw", int_t, 0, "", 100, 0, 200)
gen.add("max_thrust", double_t, 0, "", 100, 0, 200)
gen.add("max_bodyrate", double_t, 0, "", 0.5, 0, 2.0)

# gen.add("MPC_VIEW_CTL",   bool_t,   0, "",  False)

gen.add("PX4_FEED_THROUGH",   bool_t,   0, "",  False)
gen.add("goal_thrust", double_t, 0, "", 0.02, 0.0, 1.0)

gen.add("enable_state_change",   bool_t,   0, "",  False)
gen.add("boot",   bool_t,   0, "",  False)
gen.add("emergency_land",   bool_t,   0, "",  False)

exit(gen.generate(PACKAGE, "autopilot_monitor", "BasicTune"))
